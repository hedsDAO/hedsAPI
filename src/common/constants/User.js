export const DEFAULT_PROFILE_PICTURE =
  'https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/users%2F0x000000000000000000000000000000.png?alt=media&token=55cb53fe-736d-4b1e-bcd0-bf17bc7146dc';
