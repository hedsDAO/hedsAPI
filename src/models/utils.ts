import { User } from './common';

export const emptyUserState = (state: User): User => {
  return {
    ...state,
    profilePicture: '',
    twitterHandle: '',
    badges: [],
    description: '',
    displayName: '',
    role: 0,
    wallet: '',
    samples: {},
    tracks: {},
    submissions: {},
    collection: {},
    public: false,
  };
};
