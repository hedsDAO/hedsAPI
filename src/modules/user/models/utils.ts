import { User } from './common';

/**
 * @name emptyUserState
 * @summary Function to clear and return a typed user state.
 *
 * @note state is recieved as an argument, spread and overwritten to
 * prevent unused variable ts error in UserModal.
 */

export const emptyUserState = (state: User): User => {
  return {
    profilePicture: '',
    twitterHandle: '',
    badges: [],
    description: '',
    displayName: '',
    role: 0,
    wallet: '',
    samples: {},
    tracks: {},
    submissions: {},
    public: false,
  };
};
