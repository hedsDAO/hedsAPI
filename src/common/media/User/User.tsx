import { User as UserMetadata } from '@/models/common';
import { Avatar, Box, Flex, Stack } from '@chakra-ui/react';
import { formatWallet } from '@/utils';

const User = ({ user, size }: { user: UserMetadata; size?: 'sm' | 'lg' }) => {
  return (
    <>
      {size === 'lg' || !size ? (
        <Flex mt={'0 !important'}>
          <Box bg="">
            <Flex>
              <Avatar size="md" border={'solid 1px'} src={user.profile_picture} />
              <Stack justifyContent={'center'} pl={4}>
                <Flex alignItems={'center'} fontSize="2xs" gap={2}>
                  <Box fontFamily={'inter'} letterSpacing={'widest'} fontSize={'sm'} mt={'0 !important'}>
                    {user.display_name}
                  </Box>
                  <i className="fa-regular fa-arrow-up-right-from-square"></i>
                </Flex>
                <Box color="blackAlpha.400" fontFamily={'mono'} letterSpacing={'widest'} fontSize={'2xs'} mt={'0 !important'}>
                  {formatWallet(user.wallet)}
                </Box>
              </Stack>
            </Flex>
          </Box>
        </Flex>
      ) : (
        <Stack w='full' pt={2} alignItems={'center'}>
          <Avatar size="xl" src={user.profile_picture} />
          <Box letterSpacing={'widest'} fontSize={'xs'} mt={'2 !important'}>
            {user.display_name}
          </Box>
          <Box letterSpacing={'widest'} fontSize={'xs'} color={'blackAlpha.700'} mt={'0 !important'}>
            {formatWallet(user.wallet)}
          </Box>
        </Stack>
      )}
    </>
  );
};

export default User;
