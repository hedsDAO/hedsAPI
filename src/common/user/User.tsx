import { User as UserMetadata } from '@/models/common';
import { Avatar, Box, Flex, Stack } from '@chakra-ui/react';
import { formatWallet } from '@/utils';

const User = ({ user, size }: { user: UserMetadata; size?: 'sm' | 'lg' }) => {
  return (
    <>
      {size === 'lg' || !size ? (
        <Flex>
          <Avatar src={user.profile_picture} />
          <Stack justifyContent={'center'} pl={4}>
            <Box letterSpacing={'widest'} fontSize={'sm'} mt={'0 !important'}>
              {user.display_name}
            </Box>
            <Box letterSpacing={'widest'} fontSize={'sm'} color={'blackAlpha.700'} mt={'0 !important'}>
              {formatWallet(user.wallet)}
            </Box>
          </Stack>
        </Flex>
      ) : (
        <Box>
          <Avatar size="xl" src={user.profile_picture} />
          <Stack pt={2} alignItems={'center'}>
            <Box letterSpacing={'widest'} fontSize={'xs'} mt={'0 !important'}>
              {user.display_name}
            </Box>
            <Box letterSpacing={'widest'} fontSize={'xs'} color={'blackAlpha.700'} mt={'0 !important'}>
              {formatWallet(user.wallet)}
            </Box>
          </Stack>
        </Box>
      )}
    </>
  );
};

export default User;
