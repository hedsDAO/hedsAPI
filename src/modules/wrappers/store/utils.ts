import { User, BadgeData, TrackMetadata, UserCollection } from '@/models/common';
import { DateTime } from 'luxon';
import { userModelState } from './userModel';

/**
 * @function emptyUserState
 * @summary Function to clear and return a typed user state.
 * @note state is recieved as an argument, spread and overwritten to
 * prevent unused variable ts error in UserModal.
 */

export const clearUserState = (state: userModelState): userModelState => {
  const empty: User = {
    profilePicture: '',
    banner: '',
    twitterHandle: '',
    badges: [],
    description: '',
    displayName: '',
    role: 0,
    wallet: '',
    samples: {},
    tracks: {},
    submissions: {},
    collection: {} as UserCollection,
    joined: null,
  };
  return {
    connectedUser: empty,
    currentUser: empty,
    audioTabIndex: 0,
    engagementTabIndex: 0,
    audioTabs: [] as Array<string>,
    engagementTabs: [] as Array<string>,
  };
};

/**
 * @function clearConnectedUserState
 * @summary Function to clear and return a typed user state.
 * @note state is recieved as an argument, spread and overwritten to
 * prevent unused variable ts error in UserModal.
 */

export const clearConnectedUserState = (state: userModelState): userModelState => {
  const newState = { ...state };
  const currentUserState = newState.currentUser;
  return {
    currentUser: currentUserState,
    connectedUser: {
      profilePicture: '',
      banner: '',
      twitterHandle: '',
      badges: [],
      description: '',
      displayName: '',
      role: 0,
      wallet: '',
      samples: {},
      tracks: {},
      submissions: {},
      collection: {} as UserCollection,
      joined: null,
    },
    audioTabIndex: 0,
    engagementTabIndex: 0,
  };
};
export const clearCurrentUserState = (state: userModelState): userModelState => {
  const empty: User = {
    profilePicture: '',
    banner: '',
    twitterHandle: '',
    badges: [],
    description: '',
    displayName: '',
    role: 0,
    wallet: '',
    samples: {},
    tracks: {},
    submissions: {},
    collection: {} as UserCollection,
    joined: null,
  };
  const newState = { ...state };
  const connectedUserState = newState.connectedUser;
  return {
    currentUser: empty,
    connectedUser: connectedUserState,
    audioTabIndex: 0,
    engagementTabIndex: 0,
    audioTabs: [] as Array<string>,
    engagementTabs: [] as Array<string>,
  };
};

/**
 * @function populateNewUser
 * @summary populated new user data with @const {string} wallet recieved from wagmi.
 */

export const populateNewUser = (wallet: string, displayName: string): User => {
  const STORAGE_PREFIX_URL = 'https://firebasestorage.googleapis.com/v0/b/heds-104d8.appspot.com/o';
  const vistorBadge: BadgeData = {
    description: 'Welcome to heds.',
    image: `${STORAGE_PREFIX_URL}/badges%2Fvisitor.png?alt=media&token=468508bd-2831-4bd2-b943-329e5608cad1`,
    name: 'Visitor',
  };
  const newUserData = {
    profilePicture: `${STORAGE_PREFIX_URL}/profilePictures%2F0x000000000000000000000000000000.png?alt=media&token=55cb53fe-736d-4b1e-bcd0-bf17bc7146dc`,
    banner: `${STORAGE_PREFIX_URL}/banners%2F0x000000000000000000000000000000.png?alt=media&token=c2e9c947-5965-4d77-b0c3-047c2bc125d3`,
    twitterHandle: '',
    badges: [vistorBadge],
    description: '',
    displayName: displayName,
    wallet: wallet.toLowerCase(),
    collection: {} as UserCollection,
    samples: {},
    tracks: {},
    submissions: {},
    role: 0,
    joined: DateTime.now().toMillis(),
    likes: [] as TrackMetadata[],
  };
  return newUserData;
};

export default populateNewUser;
