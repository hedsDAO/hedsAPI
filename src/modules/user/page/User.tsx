import { useSelector } from 'react-redux';
import { RootState, store } from '@/store';
import { TapeData } from '@/models/common';
import { Flex, Stack } from '@chakra-ui/react';
import UserWrapper from '@/common/components/wrappers/UserWrapper';
import {
  FeaturedSubmissions,
  TwitterLinkButton,
  UserSubmissions,
  ProfilePicture,
  CopyWalletButton,
  DisplayName,
  UserDescription,
  PrivateUserWrapper,
  UserCollection,
  RefreshCollectionButton,
} from '@/modules/user/components';

export const User: React.FC = (): JSX.Element => {
  const loading = useSelector((state: RootState) => state.loading.models.userModel);
  const userData = useSelector((state: RootState) => state.userModel);
  const tapeData = useSelector((state: RootState) => state.tapesModel);
  const userTracks: { [key: string]: TapeData } = store.select.userModel.getTapeCovers(store.getState(), tapeData.allTapes);
  return (
    <Flex maxWidth={'7xl'} mx={'auto'} flexDirection={['column', 'column', 'row', 'row']} gap={10} px={[10, 4, 2, 0]} py={10}>
      <UserWrapper>
        <Stack direction={'column'}>
          <ProfilePicture loading={loading} userData={userData} />
          <DisplayName loading={loading} userData={userData} />
          <UserDescription loading={loading} userData={userData} />
          <CopyWalletButton loading={loading} wallet={userData?.wallet} />
          <TwitterLinkButton loading={loading} userData={userData} />
        </Stack>
        <Stack direction={'column'} spacing="2" width={'full'}>
          <PrivateUserWrapper loading={loading} userData={userData}>
            <UserSubmissions loading={loading} userData={userData} />
            <FeaturedSubmissions loading={loading} userTracks={userTracks} />
            <RefreshCollectionButton userData={userData} />
            <UserCollection loading={loading} userData={userData} />
          </PrivateUserWrapper>
        </Stack>
      </UserWrapper>
    </Flex>
  );
};
