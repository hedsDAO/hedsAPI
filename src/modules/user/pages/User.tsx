import React from 'react';
import { useSelector } from 'react-redux';
import { RootState } from '@/store';
import { Heading, Stack } from '@chakra-ui/react';
import UserWrapper from '@/common/components/wrappers/UserWrapper';
import { TwitterLinkButton, UserSubmissions, ProfilePicture, CopyWalletButton, DisplayName, UserDescription } from '@/modules/user/components';

export const User = () => {
  const loading = useSelector((state: RootState) => state.loading.models.userModel);
  const userData = useSelector((state: RootState) => state.userModel);
  return (
    <div className="max-w-7xl mx-auto flex md:flex-row flex-col gap-10 lg:px-0 px-4 py-10 min-h-screen">
      <UserWrapper>
        <Stack direction={'column'}>
          <ProfilePicture loading={loading} userData={userData} />
          <DisplayName loading={loading} userData={userData} />
          <UserDescription loading={loading} userData={userData} />
          <CopyWalletButton loading={loading} wallet={userData?.wallet} />
          <TwitterLinkButton loading={loading} userData={userData} />
        </Stack>
        <Stack direction={'column'} spacing="2" width={'full'}>
          <Heading fontSize={'3xl'}>Submissions</Heading>
          <UserSubmissions loading={loading} userData={userData} />
          <Heading fontSize={'3xl'}>Featured On</Heading>
        </Stack>
      </UserWrapper>
    </div>
  );
};
