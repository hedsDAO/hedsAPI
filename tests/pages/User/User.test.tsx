import React from 'react';
import { User } from '@pages/User/User';
import { store } from '@/store';
import renderer from 'react-test-renderer';
import { renderWithRematchStore } from '../../utils/testUtils';
import { act, screen, waitFor } from '@testing-library/react';
import { userData } from '../../mockData/User/UserData';
import { formatWallet } from '@utils/formatWallet';

describe('User unit', () => {
  describe('snapshot', () => {
    test('snapshot', async () => {
      await store.dispatch.userModel.setUserData(userData);
      const snapshot = renderWithRematchStore(<User />, store);
      expect(snapshot).toMatchSnapshot();
    });
  });
  describe('renders user data properly', () => {
    beforeEach(
      async () =>
        await act(() => {
          store.dispatch.userModel.setUserData(userData);
          renderWithRematchStore(<User />, store);
        }),
    );
    it('should render profile picture', () => {
      const profilePictureImage = screen.getByTestId('user-profile-picture');
      const { profilePicture } = userData;
      expect(profilePictureImage).toHaveAttribute('src', profilePicture);
    });
    it('should render display name', () => {
      const displayNameText = screen.getByTestId('user-display-name');
      const { displayName } = userData;
      expect(displayNameText).toHaveTextContent(displayName);
    });
    it('should render desciption', () => {
      const descriptionText = screen.getByTestId('user-description');
      const { description } = userData;
      expect(descriptionText).toHaveTextContent(description || '...');
    });
    it('should render twitter link if present', () => {
      const twitterLink = screen.queryByTestId('user-twitter-link');
      const twitterText = screen.queryByTestId('user-twitter-text');
      const { twitterHandle } = userData;
      if (twitterHandle) {
        expect(twitterLink).toHaveAttribute('href', `https://www.twitter.com/${twitterHandle}`);
        expect(twitterText).toHaveTextContent(`@${twitterHandle}`);
      } else expect(twitterText && twitterLink).toBeFalsy();
    });
    it('should render wallet address', () => {
      const walletText = screen.queryByTestId('user-wallet');
      const { wallet } = userData;
      expect(walletText).toHaveTextContent(formatWallet(wallet));
    });
    it('should copy wallet address to clipboard', () => {
      document.execCommand = jest.fn();
      const copyButton = screen.queryByTestId('user-wallet');
      const copyContainer = screen.queryByTestId('user-copy-container');
      act(() => copyButton.click());
      expect(copyContainer.lastChild).toHaveTextContent('copied');
      expect(document.execCommand).toHaveBeenCalledWith('copy');
      waitFor(() => expect(copyContainer.lastChild).not.toHaveTextContent('copied'));
    });
    it('should render user submissions', () => {
      const userSubmissions = screen.queryByTestId('user-submissions');
      const { submissions } = userData;
      if (!submissions) expect(userSubmissions).toBeEmptyDOMElement();
      else Object.values(submissions.heds.hedstape).map((sub) => expect(userSubmissions).toHaveTextContent(sub.track));
    });
  });
});
