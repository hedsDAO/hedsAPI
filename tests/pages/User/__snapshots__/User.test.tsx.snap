// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User unit snapshot snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="max-w-7xl my-10 mx-auto px-3 flex flex-row gap-8"
      >
        <div
          class="aspect-square sm:w-[25%]"
        >
          <div
            class="chakra-skeleton w-full h-full lg:min-h-[18rem] lg:max-h-[18rem] lg:min-w-[18rem] lg:max-w-[18rem] css-0"
          >
            <img
              alt="Alexander Lewis"
              class="object-cover w-full h-full lg:min-h-[18rem] lg:max-h-[18rem] lg:min-w-[18rem] lg:max-w-[18rem] aspect-square rounded-lg object-center"
              data-testid="user-profile-picture"
              src="https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/users%2F0xde5bc302fc6d899eab4334602dd15efb20cd745f.jpeg?alt=media&token=9f251bf5-38c8-4800-b4c4-bc9166a0ebc2"
            />
          </div>
          <div
            class="sm:mt-6 mt-4"
          >
            <h1
              class="text-4xl font-semibold tracking-tight text-gray-900 sm:text-3xl"
              data-testid="user-display-name"
            >
              Alexander Lewis
            </h1>
          </div>
          <p
            class="text-gray-500 text-sm mb-1 px-1"
            data-testid="user-description"
          >
            ...
          </p>
          <a
            data-testid="user-twitter-link"
            href="https://www.twitter.com/_alexanderlewis"
            rel="noreferrer"
            target="_blank"
          >
            <span
              class="inline-flex items-center text-xs self-start mt-4 tracking-tight text-neutral-600 px-2.5 py-0.5 rounded-lg transition-all "
              data-testid="user-twitter-text"
            >
              <i
                class="fa-brands fa-twitter mr-2 text-[0.75rem]"
              />
              @
              _alexanderlewis
            </span>
          </a>
          <div
            class="inline-flex"
            data-testid="user-copy-container"
          >
            <button
              class="inline-flex items-center text-xs self-start mt-4 tracking-tight text-neutral-600  px-2.5 py-0.5 rounded-lg"
              data-testid="user-wallet"
            >
              <i
                class="fa-solid fa-copy mr-2 text-[0.75rem]"
              />
              0xde...745f
            </button>
          </div>
        </div>
        <div
          class="sm:w-[75%]"
        >
          <h1
            class="text-3xl font-semibold tracking-tight text-gray-900 sm:text-3xl my-2"
          >
            Submissions
          </h1>
          <ul
            class="divide-y divide-gray-200"
            data-testid="user-submissions"
            role="list"
          >
            <li
              class="text-xs text-gray-600"
            >
              <a
                class="block hover:bg-gray-50"
                href="#"
              >
                <div
                  class="flex items-center px-4 py-3 sm:px-6 gap-x-2"
                >
                  <span
                    class="text-xs font-thin"
                  >
                    HT
                    5
                  </span>
                  <div
                    class="min-w-0 flex-1 sm:flex"
                  >
                    fretful dormouse
                  </div>
                  <div
                    class="mt-4 flex-shrink-0 sm:mt-0 sm:ml-5"
                  >
                    <span
                      class="mr-2"
                    >
                      01:14
                    </span>
                    <i
                      class="fa-solid fa-play"
                    />
                  </div>
                </div>
              </a>
            </li>
          </ul>
          <h1
            class="text-3xl font-semibold tracking-tight text-gray-900 sm:text-3xl my-2"
          >
            Featured On
          </h1>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="max-w-7xl my-10 mx-auto px-3 flex flex-row gap-8"
    >
      <div
        class="aspect-square sm:w-[25%]"
      >
        <div
          class="chakra-skeleton w-full h-full lg:min-h-[18rem] lg:max-h-[18rem] lg:min-w-[18rem] lg:max-w-[18rem] css-0"
        >
          <img
            alt="Alexander Lewis"
            class="object-cover w-full h-full lg:min-h-[18rem] lg:max-h-[18rem] lg:min-w-[18rem] lg:max-w-[18rem] aspect-square rounded-lg object-center"
            data-testid="user-profile-picture"
            src="https://firebasestorage.googleapis.com/v0/b/heds-34ac0.appspot.com/o/users%2F0xde5bc302fc6d899eab4334602dd15efb20cd745f.jpeg?alt=media&token=9f251bf5-38c8-4800-b4c4-bc9166a0ebc2"
          />
        </div>
        <div
          class="sm:mt-6 mt-4"
        >
          <h1
            class="text-4xl font-semibold tracking-tight text-gray-900 sm:text-3xl"
            data-testid="user-display-name"
          >
            Alexander Lewis
          </h1>
        </div>
        <p
          class="text-gray-500 text-sm mb-1 px-1"
          data-testid="user-description"
        >
          ...
        </p>
        <a
          data-testid="user-twitter-link"
          href="https://www.twitter.com/_alexanderlewis"
          rel="noreferrer"
          target="_blank"
        >
          <span
            class="inline-flex items-center text-xs self-start mt-4 tracking-tight text-neutral-600 px-2.5 py-0.5 rounded-lg transition-all "
            data-testid="user-twitter-text"
          >
            <i
              class="fa-brands fa-twitter mr-2 text-[0.75rem]"
            />
            @
            _alexanderlewis
          </span>
        </a>
        <div
          class="inline-flex"
          data-testid="user-copy-container"
        >
          <button
            class="inline-flex items-center text-xs self-start mt-4 tracking-tight text-neutral-600  px-2.5 py-0.5 rounded-lg"
            data-testid="user-wallet"
          >
            <i
              class="fa-solid fa-copy mr-2 text-[0.75rem]"
            />
            0xde...745f
          </button>
        </div>
      </div>
      <div
        class="sm:w-[75%]"
      >
        <h1
          class="text-3xl font-semibold tracking-tight text-gray-900 sm:text-3xl my-2"
        >
          Submissions
        </h1>
        <ul
          class="divide-y divide-gray-200"
          data-testid="user-submissions"
          role="list"
        >
          <li
            class="text-xs text-gray-600"
          >
            <a
              class="block hover:bg-gray-50"
              href="#"
            >
              <div
                class="flex items-center px-4 py-3 sm:px-6 gap-x-2"
              >
                <span
                  class="text-xs font-thin"
                >
                  HT
                  5
                </span>
                <div
                  class="min-w-0 flex-1 sm:flex"
                >
                  fretful dormouse
                </div>
                <div
                  class="mt-4 flex-shrink-0 sm:mt-0 sm:ml-5"
                >
                  <span
                    class="mr-2"
                  >
                    01:14
                  </span>
                  <i
                    class="fa-solid fa-play"
                  />
                </div>
              </div>
            </a>
          </li>
        </ul>
        <h1
          class="text-3xl font-semibold tracking-tight text-gray-900 sm:text-3xl my-2"
        >
          Featured On
        </h1>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
